frontend/src/metabase/dashboard/components/EmbedMenu/EmbedMenu.unit.spec.tsx

```diff
@@ -0,0 +1,158 @@
+import userEvent from "@testing-library/user-event";
+import { EmbedMenu } from "metabase/dashboard/components/EmbedMenu/EmbedMenu";
+import { renderWithProviders, screen } from "__support__/ui";
+import { createMockUser } from "metabase-types/api/mocks";
+import {
+  createMockSettingsState,
+  createMockState,
+} from "metabase-types/store/mocks";
+
+interface SetupProps {
+  isAdmin: boolean;
+  hasPublicLink?: boolean;
+  editing?: boolean;
+  embeddingEnabled?: boolean;
+  fullscreen?: boolean;
+  linkEnabled?: boolean;
+  publicLinksEnabled?: boolean;
+}
+
+// Mock embedding modal as we don't need its content for the tests and causes
+// some issues with routing for one or two tests.
+const TestEmbeddingModalComponent = () => {
+  return <div data-testid="dashboard-sharing-embedding-modal" />;
+};
+
+jest.mock(
+  "metabase/dashboard/containers/DashboardSharingEmbeddingModal",
+  () => {
+    return {
+      __esModule: true,
+      default: TestEmbeddingModalComponent,
+    };
+  },
+);
+
+const setup = ({
+  isAdmin,
+  hasPublicLink,
+  publicLinksEnabled = false,
+}: SetupProps) => {
+  const onModalOpen = jest.fn();
+  renderWithProviders(
+    <EmbedMenu
+      resource_uuid={hasPublicLink ? "mock-uuid" : null}
+      onModalOpen={onModalOpen}
+    />,
+    {
+      storeInitialState: createMockState({
+        currentUser: createMockUser({ is_superuser: isAdmin }),
+        settings: createMockSettingsState({
+          "enable-public-sharing": publicLinksEnabled,
+        }),
+      }),
+    },
+  );
+
+  return {
+    onModalOpen,
+  };
+};
+
+describe("EmbedMenu", () => {
+  describe("which label should be rendered", () => {
```

Comment:
There should be an additional test for [the new label](https://www.notion.so/metabase/Setup-flow-for-static-public-embedding-embedding-feature-discoverability-3f7362e71e63440883bfebae48ab54a0#5c55501b5ef341a7a6f6522e21606e28) when both public sharing and embedding are disabled.